src/System.Memory/src/System/Buffers/Text/Utf8Formatter/FormattingHelpers.cs

```diff
@@ -16,111 +16,242 @@ namespace System.Buffers.Text
     // code must have already done all the necessary validation.
     internal static class FormattingHelpers
     {
-        // For the purpose of formatting time, the format specifier contains room for
-        // exactly 7 digits in the fraction portion. See "Round-trip format specifier"
-        // at the following URL for more information.
-        // https://msdn.microsoft.com/en-us/library/az4se3k1(v=vs.110).aspx#Roundtrip
-        private const int FractionDigits = 7;
-
         // A simple lookup table for converting numbers to hex.
-        private const string HexTable = "0123456789abcdef";
+        internal const string HexTableLower = "0123456789abcdef";
 
-        #region UTF-8 Helper methods
+        internal const string HexTableUpper = "0123456789ABCDEF";
+
+        // TODO: Where should the below method live? Ideally it should be publicly exposed
+        // since having blittable access to structs would be convenient.
+        // Should it be on BinaryPrimitives?
 
+        /// <summary>
+        /// Given a reference to a blittable value type, returns a Span that allows
+        /// access to the binary representation of the value.
+        /// </summary>
+        /// <remarks>
+        /// No copy is performed as part of this method's logic.
+        /// This is intended to be a "safe" API even though it's implemented in terms of unsafe casts.
+        /// </remarks>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public static void WriteHexByte(byte value, ref byte buffer, int index)
+        public static Span<byte> GetSpanForBlittable<T>(ref T value) where T : struct
         {
-            Unsafe.Add(ref buffer, index) = (byte)HexTable[value >> 4];
-            Unsafe.Add(ref buffer, index + 1) = (byte)HexTable[value & 0xF];
+#if netstandard
+            if (SpanHelpers.IsReferenceOrContainsReferences<T>())
+            {
+                ThrowHelper.ThrowArgumentException_InvalidTypeWithPointersNotSupported(typeof(T));
+            }
+#else
+            if (RuntimeHelpers.IsReferenceOrContainsReferences<T>())
+            {
+                ThrowHelper.ThrowArgumentException_InvalidTypeWithPointersNotSupported(typeof(T));
+            }
+#endif
+            return Span<byte>.DangerousCreate(null, ref Unsafe.As<T, byte>(ref value), Unsafe.SizeOf<T>());
```

Comment:
This is not valid - it crash with slow Span. 

It is not valid to pass null as first argument to DangerousCreate,

Reply:
Previous closed issue about this https://github.com/dotnet/corefx/issues/26124